enterprise/server/remote_execution/commandutil/commandutil_windows.go

```diff
@@ -0,0 +1,89 @@
+//go:build windows
+
+package commandutil
+
+import (
+	"fmt"
+	"os/exec"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+type process struct {
+	cmd        *exec.Cmd
+	terminated chan struct{}
+
+	jobHandle windows.Handle
+}
+
+func (p *process) preStart() error {
+	j, err := createWindowsJobObject()
+	if err != nil {
+		return err
+	}
+
+	p.jobHandle = j
+	return nil
+}
+
+func (p *process) postStart() error {
+	processHandle, err := windows.OpenProcess(windows.PROCESS_QUERY_LIMITED_INFORMATION|windows.PROCESS_TERMINATE, false, uint32(p.cmd.Process.Pid))
+	if err != nil {
+		return nil
+	}
+	defer windows.CloseHandle(processHandle)
+
+	return windows.AssignProcessToJobObject(p.jobHandle, processHandle)
```

Comment:
does this work if the process has exited by the time we get here? (can test this by adding a 100 ms delay before calling this, and running a NOP process - whatever the `sleep 0` equivalent is on Windows or something)

I wonder if there is a cleaner way to do this, like assign the process to the job at creation time or something

Reply:
This is a really good call.

I tested it and it failed with sleep.

I think the correct way to do this would be to create a suspended process, assign it to a job object, and then resume the process. https://devblogs.microsoft.com/oldnewthing/20130405-00/?p=4743
Let me look into fixing it.

Reply:
It's fixed